components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/ConfigPersistenceManagerTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.entitlement.persistence;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.database.utils.jdbc.NamedPreparedStatement;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.entitlement.EntitlementException;
+import org.wso2.carbon.identity.entitlement.cache.ConfigCache;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+import static org.testng.Assert.*;
+import static org.wso2.carbon.identity.entitlement.PDPConstants.Algorithms.*;
+import static org.wso2.carbon.identity.entitlement.PDPConstants.Algorithms.ORDERED_PERMIT_OVERRIDES;
+
+/**
+ * This is the parent test class for the Config Persistence Manager test classes.
+ */
+public abstract class ConfigPersistenceManagerTest {
+
+    ConfigPersistenceManager configPersistenceManager;
+
+    @Test
+    public void testGetDefaultGlobalPolicyAlgorithmName() {
+
+        String globalPolicyAlgorithmName = configPersistenceManager.getGlobalPolicyAlgorithmName();
+        assertEquals(globalPolicyAlgorithmName, DENY_OVERRIDES);
+    }
+
+    @DataProvider
+    public Object[][] globalPolicyAlgorithmData() {
+
+        return new Object[][]{
+                {DENY_OVERRIDES},
+                {PERMIT_OVERRIDES},
+                {FIRST_APPLICABLE},
+                {ONLY_ONE_APPLICABLE},
+                {ORDERED_DENY_OVERRIDES},
+                {ORDERED_PERMIT_OVERRIDES}
+        };
+    }
+
+    @Test(dataProvider = "globalPolicyAlgorithmData")
+    public void testAddGlobalPolicyAlgorithm(String policyCombiningAlgorithm) throws Exception {
+
+        // Add the first global policy combining algorithm.
+        configPersistenceManager.addOrUpdateGlobalPolicyAlgorithm(policyCombiningAlgorithm);
+        String policyAlgorithmNameFromDb = configPersistenceManager.getGlobalPolicyAlgorithmName();
```

Comment:
Better to use name like `policyAlgorithmNameFromStorage`

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/5905/commits/fe59d644b547388f83603a9ac1a0e90effde1ec4